### PR TITLE
Make evp_test skip mac tests if digest or ciphers are disabled.

### PR DESCRIPTION
**Pull request by slontis**
Fixes test error in #18714
This only happens currently during minimal builds.

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by slontis** [comment]
tested locally with the
config options used by a minimal build and data from #18714

**Review by paulidale — APPROVED** [review]

File: test/evp_test.c [hunk 1]
```diff
@@ -1444,25 +1444,39 @@ static int mac_test_run_mac(EVP_TEST *t)
...
-                OSSL_PARAM_construct_utf8_string(OSSL_MAC_PARAM_CIPHER,
-                                                 expected->alg, 0);
-        } else if (OSSL_PARAM_locate_const(defined_params,
+            if (is_cipher_disabled(expected->alg))
+                skip = 1;
+            else
+                params[params_n++] =
+                    OSSL_PARAM_construct_utf8_string(OSSL_MAC_PARAM_CIPHER,
+                                                     expected->alg, 0);
+            } else if (OSSL_PARAM_locate_const(defined_params,
```
Comment (hlandau) [inline test/evp_test.c:1462]: Please indent this else if clause correctly

**Review by hlandau — CHANGES_REQUESTED** [review]

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
IMO OK for 3.0 as well. @paulidale do you agree?

**Comment by paulidale** [comment]
Good for 3.0 too.

File: test/evp_test.c (hunk 1 above)
Comment (slontis) [reply test/evp_test.c:1462]: Thanks for spotting..

**Review by paulidale — APPROVED** [review]

**Comment by paulidale** [comment]
Merged.

**Pull request closed without merging**